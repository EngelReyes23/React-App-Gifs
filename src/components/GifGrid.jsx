import React from "react";
import PropTypes from "prop-types";
import { useFetchGif } from "../hooks/useFetchGif";
import GifGridItems from "./GifGridItems";
import Spinner from "./Spinner";

const GifGrid = ({ category }) => {
	const { data: imgs, loading } = useFetchGif(category);

	return (
		<>
			<h3>{category}</h3>
			{loading && <Spinner />}

			<div className="card-grid">
				{imgs.length !== 0 ? (
					imgs.map((img) => <GifGridItems {...img} key={img.id} />)
				) : (
					<p>No hay resultados</p>
				)}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};

export default GifGrid;
