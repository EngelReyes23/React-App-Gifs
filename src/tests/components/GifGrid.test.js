import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGif } from "../../hooks/useFetchGif";
jest.mock("../../hooks/useFetchGif");

describe("Pruebas a <GifGrid />", () => {
	// instacia de wrapper
	let wrapper;
	beforeEach(() => {
		useFetchGif.mockReturnValue({
			data: [],
			loading: true,
		});
		wrapper = shallow(<GifGrid category="Goku" />);
	});

	// 1ra prueba
	test("Debe de mostrar el componente correctamente", () => {
		//
		expect(wrapper).toMatchSnapshot();
	});

	// 2da prueba
	test("debe de mostrar items cuando se cargan imagenes useFetchGif", () => {
		const gifs = [
			{
				id: "ABC",
				url: "https://localhost/cualquiercosa.jpg",
				title: "Cualquier cosa",
			},
			{
				id: "ABC2",
				url: "https://localhost/cualquiercosa.jpg",
				title: "Cualquier cosa2",
			},
		];
		useFetchGif.mockReturnValue({
			data: gifs,
			loading: false,
		});
		const wrapper = shallow(<GifGrid category="Goku" />);
		expect(wrapper).toMatchSnapshot();

		// verifica que el loader del parrafo exista
		expect(wrapper.find("p").exists()).toBe(false);

		// verifica que el componente GifGrid exista
		expect(wrapper.find("GifGridItems").length).toBe(gifs.length);
	});
});
