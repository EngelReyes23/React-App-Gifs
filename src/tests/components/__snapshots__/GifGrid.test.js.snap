// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <Spinner />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas a <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGif 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItems
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
    <GifGridItems
      id="ABC2"
      key="ABC2"
      title="Cualquier cosa2"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
